manifests/login_defs.pp

```diff
@@ -0,0 +1,86 @@
+  Boolean                               $log_unkfail_enab      = true,
+  Boolean                               $mail_check_enab       = true,
+  Stdlib::AbsolutePath                  $mail_dir              = '/var/spool/mail',
+  Optional[Stdlib::AbsolutePath]        $mail_file             = undef,
+  Optional[Integer]                     $max_members_per_group = undef,
+  Optional[Array[Stdlib::AbsolutePath]] $motd_file             = undef,
+  Optional[Stdlib::AbsolutePath]        $nologins_file         = undef,
+  Boolean                               $obscure_checks_enab   = true,
+  Boolean                               $pass_always_warn      = true,
+  Integer                               $pass_change_tries     = 3,
+  Integer                               $pass_max_days         = 180,  # CCE-26985-2
+  Integer                               $pass_min_days         = 1,    # CCE-27013-2
+  Integer                               $pass_warn_age         = 14,   # CCE-26998-6
+  Optional[Integer]                     $pass_max_len          = undef,
+  Integer                               $pass_min_len          = 14,   # CCE-27002-5
+  Boolean                               $porttime_checks_enab  = true,
+  Boolean                               $quotas_enab           = true,
+  Integer                               $sha_crypt_min_rounds  = 5000,
+  Integer                               $sha_crypt_max_rounds  = 10000,
+  Optional[Stdlib::AbsolutePath]        $sulog_file            = undef,
+  String                                $su_name               = 'su',
+  Boolean                               $su_wheel_only         = false,
+  Optional[Integer]                     $sys_gid_max           = undef,
+  Optional[Integer]                     $sys_gid_min           = undef,
+  Optional[Integer]                     $sys_uid_max           = undef,
+  Optional[Integer]                     $sys_uid_min           = undef,
+  Boolean                               $syslog_sg_enab        = true,
+  Boolean                               $syslog_su_enab        = true,
+  Optional[String]                      $ttygroup              = undef,
+  Optional[String]                      $ttyperm               = undef,
+  Optional[Stdlib::AbsolutePath]        $ttytype_file          = undef,
+  Integer                               $uid_max               = 1000000,
+  Optional[Integer]                     $uid_min               = undef,
+  String                                $umask                 = '007',  # CCE-26371-5
+  Optional[Integer]                     $ulimit                = undef,  # The maximum file size in 512 byte units. Noted here since the man page isn't helpful.
+  Optional[Stdlib::AbsolutePath]        $userdel_cmd           = undef,
+  Boolean                               $usergroups_enab       = true
+) {
+
+  validate_re($chfn_restrict,'^[frwh]+$')
+  if !empty($ttyperm) { validate_umask($ttyperm) }
```

Comment:
This should simply be validate_umask($ttypperm), as login_defs.erb only checks for the nil? case not the empty case.  In other words, login_defs.erb will put out a malformed line if this is empty.